client/devdocs/design/search-collection.jsx

```diff
@@ -0,0 +1,98 @@
+/**
+* External dependencies
+*/
+import React from 'react';
+
+const Hider = React.createClass( {
+	displayName: 'Hider',
+
+	propTypes: {
+		hide: React.PropTypes.bool,
+	},
+
+	shouldComponentUpdate( nextProps ) {
+		return this.props.hide !== nextProps.hide;
+	},
+
+	render() {
+		return (
+			<div style={ this.props.hide ? { display: 'none' } : {} }>
+				{ this.props.children }
+			</div>
+		);
+	}
+} );
+
+const FilterSummary = React.createClass( {
+	render() {
+		let names;
+
+		if ( this.props.items.length === 0 ) {
+			return ( <p>No matches found</p> );
+		} else if ( this.props.items.length === this.props.total || this.props.items.length === 1 ) {
+			return null;
+		}
+
+		names = this.props.items.map( function( item ) {
+			return item.props.children.type.displayName;
+		} );
+
+		return (
+			<p>Showing: { names.join( ', ' ) }</p>
+		);
+	}
+} );
+
+export default React.createClass( {
+	displayName: 'Collection',
+
+	shouldWeHide: function( example ) {
+		let filter, searchString;
+
+		filter = this.props.filter || '';
+		searchString = example.type.displayName;
+
+		if ( this.props.component ) {
+			return example.type.displayName.toLowerCase() !== this.props.component.replace( /-([a-z])/g, '$1' );
+		}
+
+		if ( example.props.searchKeywords ) {
+			searchString += ' ' + example.props.searchKeywords;
+		}
+
+		return ! ( ! filter || searchString.toLowerCase().indexOf( filter ) > -1 );
```

Comment:
@aduth I believe that is related to the new HappinessSupport component
recently added to the app components page.

On Wednesday, March 9, 2016, Andrew Duthie <notifications@github.com> wrote:

> In client/devdocs/design/search-collection.jsx
> <https://github.com/Automattic/wp-calypso/pull/3813#discussion_r55561772>:
>
> > +
> > +	shouldWeHide: function( example ) {
> > +		let filter, searchString;
> > +
> > +		filter = this.props.filter || '';
> > +		searchString = example.type.displayName;
> > +
> > +		if ( this.props.component ) {
> > +			return example.type.displayName.toLowerCase() !== this.props.component.replace( /-([a-z])/g, '$1' );
> > +		}
> > +
> > +		if ( example.props.searchKeywords ) {
> > +			searchString += ' ' + example.props.searchKeywords;
> > +		}
> > +
> > +		return ! ( ! filter || searchString.toLowerCase().indexOf( filter ) > -1 );
>
> Seeing an error on this line when trying to search on the App Components
> screen:
>
> Uncaught TypeError: Cannot read property 'toLowerCase' of undefined
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Automattic/wp-calypso/pull/3813/files#r55561772>.
>